File: src/main/java/com/cuupa/mailprocessor/services/SemanticResult.java
package com.cuupa.mailprocessor.services;

import java.io.Serializable;
import java.util.List;

public class SemanticResult implements Serializable {

    private final String topicName;

    private String sender;

    private List<Metadata> metaData;

    public SemanticResult(String topicName) {
        this.topicName = topicName;
    }

    public String getTopicName() {
        return topicName;
    }

    public void setMetaData(List<Metadata> metaData) {
        this.metaData = metaData;
    }

    public List<Metadata> getMetaData() {
        return metaData;
    }

    public String getSender() {
        return sender;
    }

    public void setSender(String sender) {
        this.sender = sender;
    }

}

File: src/main/java/com/cuupa/mailprocessor/services/semantic/SemanticResult.kt
package com.cuupa.mailprocessor.services.semantic

import org.apache.commons.lang3.builder.ToStringBuilder
import java.io.Serializable

class SemanticResult(val topicName: String, val sender: String) : Serializable {

    var metaData: MutableList<Metadata> = mutableListOf()

    override fun toString(): String {
        return ToStringBuilder(this).append("topicName", topicName)
                .append("sender", sender)
                .append("metaData", metaData)
                .toString()
    }

}